Add Menu button in new header variant (#3478)
ONSdigital/design-system: src/components/header/_macro-options.md

| titleUrl              | string                                            | false                                                       | Wraps the title logo in a link. Set the URL for the HTML `href` attribute for the link.                                                        |
| button                | object`<SignOutButton>`                           | false                                                       | Settings for the [sign out button](#signoutbutton) in the header used to exit a transactional service                                          |
| navigation            | array`<Navigation>`                               | false                                                       | Settings for the [main menu links](#navigation)                                                                                                |
| siteSearchAutosuggest | `Autosuggest` [_(ref)_](/components/autosuggest)  | false                                                       | Sets the autosuggest functionality in the header                                                                                               |

> | navLinks              | object`<ServiceLinks>`                            | false                                                       | Settings for the [menu button navigation](#navLinks) in the masthead                                                                           |

Reviewer: I think nav links is quite close to navigation, we need to differentiate between them because they are different things. Also the object that gets passed in needs to match the name of the object in the documentation (ServiceLinks isn't the object)

Author: Renamed it to menuNavLinks.

## NavLinks

| Name            | Type                   | Required | Description                                                                                                                           |
| --------------- | ---------------------- | -------- | ------------------------------------------------------------------------------------------------------------------------------------- |
| id              | string                 | true     | The HTML `id` of the `<nav>` element. Used for the `aria-controls` attribute for the menu toggle button displayed on small viewports. |
| classes         | string                 | false    | Classes to add to the `<nav>` element                                                                                                 |
| ariaLabel       | string                 | false    | The `aria-label` attribute added to the `<nav>` element. Defaults to “Menu links navigation”.                                         |
| ariaListLabel   | string                 | false    | The `aria-label` attribute added to the `<ul>` element. Defaults to “Menu links”.                                                     |

> | keyLinksList    | array`<Item>`          | true     | Settings for an array of [key list items](#keyitem) for each navigation link                                                          |

Reviewer: Think this should just be `keyLinks`

> | keyLinks        | array`<KeyItem>`       | true     | Settings for an array of [key list items](#keyitem) for each navigation link                                                          |

Reviewer: ```suggestion
| keyLinks        | array`<KeyLink>`       | true     | Settings for an array of [key links](#keyLink) which will be the key highlighted links on the menu                                    |
```

> | itemsList       | array`<Item>`          | true     | Settings for an array of [list items](#item) for each navigation link                                                                 |

Reviewer: Shouldnt this be linksList? But I actually think this isn't descriptive enough. Maybe it should be
menuLinks > menuGroup > Items I think thats a bit more descriptive to the user what that is referring to

Reviewer: Looking at it I think we can rename "menuGroup" to "menuCol" and then the "subItems" could be "groupItems". Then we would have the menu columns named as such and the same for the groups within them. I think this makes more sense to me because it relates more to what is on screen

> | columns         | array`<Columns>`       | true     | Settings for list of [columns](#column) that contain menu links                                                                       |

Reviewer: ```suggestion
| columns         | array`<Column>`       | true     | Settings for list of [columns](#column) that contain menu links                                                                       |
```

> | menuLinks       | array`<menuGroup>`     | true     | Settings for an array of [list items](#item) for each navigation link                                                                 |

Reviewer: ```suggestion
| menuLinks       | array`<MenuGroup>`     | true     | Settings for an array of [list items](#item) for each navigation link                                                                 |
```

> | menuLinks       | array`<menuCol>`       | true     | Settings for an array of [list items](#menucol) for each navigation link                                                              |

Reviewer: The description for this should say that its the config for a list of columns for the menu that contains the menu links

Author: Is this one okay? 'Settings for list of [columns](#column) that contain menu links'

| itemsList              | array`<Item>`             | true     | Settings for an array of [list items](#item) for each navigation link                                                          |
| currentPath            | string or array`<string>` | true     | The path of the current active page. Multiple paths can be provided using an array to highlight nested navigation.             |
| removeHorizontalSubNav | boolean                   | false    | Set to “true” to remove the sub navigation                                                                                     |

> ## menuGroup

Reviewer: ```suggestion
## MenuGroup
```

| itemsList              | array`<Item>`             | true     | Settings for an array of [list items](#item) for each navigation link                                                          |
| currentPath            | string or array`<string>` | true     | The path of the current active page. Multiple paths can be provided using an array to highlight nested navigation.             |
| removeHorizontalSubNav | boolean                   | false    | Set to “true” to remove the sub navigation                                                                                     |

> ## KeyItem

Reviewer: ```suggestion
## KeyLink
```

| Name        | Type   | Required | Description                                                           |
| ----------- | ------ | -------- | --------------------------------------------------------------------- |
| heading     | string | true     | The heading for the item                                              |
| url         | string | false    | The URL for the HTML `href` attribute for the path to the linked page |

> | description | string | true     | The description label for the item                                    |

Reviewer: ```suggestion
| description | string | true     | The description label for the link                                    |
```

## menuCol

| Name       | Type               | Required | Description                                                           |
| ---------- | ------------------ | -------- | --------------------------------------------------------------------- |
| heading    | string             | true     | The heading label for the menu group                                  |
| url        | string             | false    | The URL for the HTML `href` attribute for the path to the linked page |

> | groupItems | array`<groupItem>` | false    | Settings for an array of [sub items](#groupItem) for each list item   |

Reviewer: ```suggestion
| groupItems | array`<groupItem>` | false    | Settings for an array of [sub items](#groupitem) for each list item   |
```

| itemsList              | array`<Item>`             | true     | Settings for an array of [list items](#item) for each navigation link                                                          |
| currentPath            | string or array`<string>` | true     | The path of the current active page. Multiple paths can be provided using an array to highlight nested navigation.             |
| removeHorizontalSubNav | boolean                   | false    | Set to “true” to remove the sub navigation                                                                                     |

## menuGroup

| Name     | Type             | Required | Description                                                           |
| -------- | ---------------- | -------- | --------------------------------------------------------------------- |
| text     | string           | true     | The text label for the link                                           |
| url      | string           | false    | The URL for the HTML `href` attribute for the path to the linked page |
| children | array`<SubItem>` | false    | Settings for an array of [sub items](#subitem) for each list item     |

> ## KeyItem

Reviewer 2: Might need to get some clarification on this but the ticket says: 
_"Top section can contain up to 3 columns. User can add 1, 2 or 3 columns. Each column have a header and **under it contains hyperlinks**."_ 

And goes on to say:
_"Lower section can contain up to 3 columns. Each column can have up to 3 sub-sections. There is no limit in the number of lines in each column and there is not constrain on the number of digits on each line.  Sub-sections will have a header and **under it hyperlinks**."_

Because of that, I expected that items underneath the headings in keylinks to be linkable, rather than the headings itself. So for example, in menuLinks, the lines under the headings are linkable
![image](https://github.com/user-attachments/assets/95385838-531c-46d5-b03a-6719e12b32d1)

Whereas in keylinks the headings themselves are linkable
![image](https://github.com/user-attachments/assets/3a0d1db1-95b2-488f-b62c-6732d5667d22)

I feel these should work the same, but that's my interpretation of the ticket requirements

Reviewer: Can you double check this with Joe? Because don't have an example of which way it should be

Author: I had a chat with Joe. KeyLinks heading will be links and the text under it will be plain text. Headings for menuLinks can be also be links.Those links take you to the parent page with all the child links in it.

| currentPath            | string or array`<string>` | true     | The path of the current active page. Multiple paths can be provided using an array to highlight nested navigation.             |
| removeHorizontalSubNav | boolean                   | false    | Set to “true” to remove the sub navigation                                                                                     |

## KeyLink

| Name        | Type   | Required | Description                                                           |
| ----------- | ------ | -------- | --------------------------------------------------------------------- |
| heading     | string | true     | The heading for the link                                              |
| url         | string | false    | The URL for the HTML `href` attribute for the path to the linked page |
| description | string | true     | The description label for the link                                    |

## Column

| Name   | Type           | Required | Description                                       |
| ------ | -------------- | -------- | ------------------------------------------------- |

> | groups | array`<group>` | true     | settings for array of [groups](#group) in column. |

Reviewer: ```suggestion
| groups | array`<group>` | true     | settings for array of [groups](#group) in column  |
```

> | Name   | Type           | Required | Description                                       |
> | ------ | -------------- | -------- | ------------------------------------------------- |
> | groups | array`<group>` | true     | settings for array of [groups](#group) in column. |

Reviewer: ```suggestion
| Name   | Type           | Required | Description                                      |
| ------ | -------------- | -------- | ------------------------------------------------ |
| groups | array`<group>` | true     | settings for array of [groups](#group) in column |
```

> ## group

Reviewer: ```suggestion
## Group
```

| Name       | Type               | Required | Description                                                           |
| ---------- | ------------------ | -------- | --------------------------------------------------------------------- |
| heading    | string             | true     | The heading label for the menu group                                  |

> | url        | string             | false    | The URL for the HTML `href` attribute for the path to the linked page |

Reviewer: Do we have a url param here?

Author: Yep, Joe told to add it for headings as well. Those will the parent pages and the ones in groupItems will be child urls

> | groupItems | array`<groupItem>` | false    | Settings for an array of [sub items](#groupitem) for each list item   |

Reviewer 2: I think [sub items] would be better renamed to [group items] for clarity
```suggestion
| groupItems | array`<groupItem>` | false    | Settings for an array of [group items](#groupitem) for each list item   |
```

> ## Groupitem

Reviewer: ```suggestion
## GroupItem
```
